src/controllers/transfer/transfer.ts

```diff
@@ -289,6 +366,62 @@ export class TransferController extends EventEmitter {
     this.checkIsRecipientAddressUnknown()
   }
 
+  #setAmount(fieldValue: string) {
+    if (!fieldValue) {
+      this.amount = ''
+      this.amountInFiat = ''
+      return
+    }
+
+    const tokenPrice = this.selectedToken?.priceIn.find(
+      (p) => p.baseCurrency === HARD_CODED_CURRENCY
+    )?.price
+
+    if (!tokenPrice) {
+      this.amount = fieldValue
+      this.amountInFiat = ''
+      return
+    }
+
+    if (this.amountFieldMode === 'fiat' && this.selectedToken?.decimals) {
+      this.amountInFiat = fieldValue
+
+      // Get the number of decimals
+      const amountInFiatDecimals = fieldValue.split('.')[1]?.length || 0
+      const { tokenPriceBigInt, tokenPriceDecimals } = convertTokenPriceToBigInt(tokenPrice)
+
+      // Convert the numbers to big int
+      const amountInFiatBigInt = parseUnits(fieldValue, amountInFiatDecimals)
```

Comment:
Kolega, that's a really good approach using Strings.

Relying on Strings means fieldValue is a String. Otherwise, if it were a number, we would face floating-point issues.